utils/process/executor.cpp

```diff
@@ -689,6 +689,34 @@ struct utils::process::executor::executor_handle::impl : utils::noncopyable {
                 data._pimpl->state_owners,
                 all_exec_handles)));
     }
+
+    executor::exit_handle
+    reap(const int original_pid)
+    {
+        const exec_handles_map::iterator iter = all_exec_handles.find(
```

Comment:
Error handling is missing for values like `0`, `-1`, etc. These values have special meanings and can result in the process either committing suicide OR a system-wide reboot/shutdown/etc. Neither case is desired.
From kill(2):
```
     If pid is zero:
             The sig signal is sent to all processes whose group ID is equal
             to the process group ID of the sender, and for which the process
             has permission; this is a variant of killpg(2).

     If pid is -1:
             If the user has super-user privileges, the signal is sent to all
             processes excluding system processes (with P_SYSTEM flag set),
             process with ID 1 (usually init(8)), and the process sending the
             signal.  If the user is not the super user, the signal is sent to
             all processes which the caller has permissions to, excluding the
             process sending the signal.  No error is returned if any process
             could be signaled.
```

Reply:
It looks the function name is misleading, it's not the same reaping as in the general process management context. During my work on this patch back in Oct, I found a defect in the existing Kyua mechanism of abrupt termination handling -- it was not collecting all exit handles of its children. It was easier to reproduce with parallelism>1, and the parallelism is what I was actively working on. The pid passed here as a parameter is a pid of a spawn child previously saved by kyua, the kyua simply expects to see an exit handle object per each child spawn and this "kyua specifics" reaping process helps with that. That is, it's named after the same term of the general context, but this is about kyua internals.

Reply:
What I was driving at before is that some invariants should be added to make sure -1/0 are never passed to this function.

Reply:
The follow-up: https://github.com/freebsd/kyua/pull/231